### ISSUE TITLE
Table erratic row selection with virtual scroll

### ISSUE THREAD

**Issue by omnight** [primefaces/primeng#8952]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Current behavior**
With virtual scroll and selectionMode=single enabled, selecting a row and then scrolling in the lists makes the highlight jump around and rows that are not selected show as highlighted.

**Expected behavior**
Only actually selected row should have ui-state-highlight

**Minimal reproduction of the problem with instructions**

https://stackblitz.com/edit/primeng-table-virtual-scroll-selection
Select a row, then scroll.

**Comment by Wanchai** [comment]
Is there a quick fix for this? Because it is really annoying...

**Comment by pinolombardo** [comment]
Are there any news about this bug?

**Comment by mauriziocolica** [comment]
I'm facing the same issue: are there any news??

**Comment by curciot** [comment]
I've got the same problem, can you give us a workaround or a fix?

**Comment by szyzu** [comment]
I had this problem too. This is my workaround:
![Nagrywaj-2020-07-30-20-24-42-285](https://user-images.githubusercontent.com/68990952/88960472-d8619f00-d2a3-11ea-84f6-d8992ddf806c.gif)

https://stackblitz.com/edit/primeng-table-virtual-scroll-workaround
The solution does not work on stackblitz. It requires Ivy to be enabled.

**Comment by GretchenGit** [comment]
Would be great to have this resolved without an extra workaround. Although szyzus solution helps for the moment

**Comment by thimma11** [comment]
I'm having the same problem with selectionMode=multiple and using checkboxes for selection. Unfortunately the fix from @szyzu does not work for me. Would be great to have this fixed.

**Comment by ryanjvdm** [comment]
Just ran into the same issue.  A fix would be appreciated.

**Comment by matteoveraldi** [comment]
It would be nice to have it been fixed without the use of a workaround.
Any news about a fix?

**Comment by curciot** [comment]
> I'm having the same problem with selectionMode=multiple and using checkboxes for selection. Unfortunately the fix from @szyzu does not work for me. Would be great to have this fixed.

i used that workaround for selectionMode=multiple and it's working but i had to debug to find that index in ngContext:

 const dataKeyValue = ngContext[24][this.dataKey];

```
private updateRowHighlight(){
    const elements = document.getElementsByClassName('ui-selectable-row');
    for (let i = 0 ; i < elements.length; i++){
      const ngContext = elements[i]['__ngContext__'];
      const dataKeyValue = ngContext[24][this.dataKey];
        if( this.entitySelected.findIndex(a=>a[this.dataKey]==dataKeyValue)==-1){
          elements[i].classList.remove('ui-state-highlight');
        }
        else{
          elements[i].classList.add('ui-state-highlight');
        }
    }
  }
```

**Comment by mattveraldi** [comment]
> > I'm having the same problem with selectionMode=multiple and using checkboxes for selection. Unfortunately the fix from @szyzu does not work for me. Would be great to have this fixed.
> 
> i used that workaround for selectionMode=multiple and it's working but i had to debug to find that index in ngContext:
> 
> const dataKeyValue = ngContext[24][this.dataKey];
> 
> ```
> private updateRowHighlight(){
>     const elements = document.getElementsByClassName('ui-selectable-row');
>     for (let i = 0 ; i < elements.length; i++){
>       const ngContext = elements[i]['__ngContext__'];
>       const dataKeyValue = ngContext[24][this.dataKey];
>         if( this.entitySelected.findIndex(a=>a[this.dataKey]==dataKeyValue)==-1){
>           elements[i].classList.remove('ui-state-highlight');
>         }
>         else{
>           elements[i].classList.add('ui-state-highlight');
>         }
>     }
>   }
> ```

We need a solution to this asap, hope that developers will catch us up with some news about it

**Comment by jadhavmanojs2019** [comment]
> I'm having the same problem with selectionMode=multiple and using checkboxes for selection. Unfortunately the fix from @szyzu does not work for me. Would be great to have this fixed.

 Is team is this issue fixed or not?

**Comment by alexandregyn** [comment]
> > I'm having the same problem with selectionMode=multiple and using checkboxes for selection. Unfortunately the fix from @szyzu does not work for me. Would be great to have this fixed.
> 
> Is team is this issue fixed or not?

yes it has been solved and is in the LTS 9.1.5 version

**Comment by git-jano** [comment]
in manual fix I recommend use debounceTime(10) when subscribing to scroll  to handle manual scroller drag&drop / pull correctly
